File: remote/src/main/java/com/kelsos/mbrc/annotations/LastfmState.java
package com.kelsos.mbrc.annotations;

import android.support.annotation.StringDef;

import java.lang.annotation.Retention;
import java.lang.annotation.RetentionPolicy;

public class LastfmState {
  public static final String LOVE = "Love";
  public static final String BAN = "Ban";
  public static final String NORMAL = "Normal";
  private LastfmState() {
    //no instance
  }

  @Retention(RetentionPolicy.SOURCE)
  @StringDef({
                 BAN,
                 LOVE,
                 NORMAL
             })
  public @interface State {
  }
}

File: app/src/main/kotlin/com/kelsos/mbrc/annotations/LastfmState.kt
package com.kelsos.mbrc.annotations

import android.support.annotation.StringDef

object LastfmState {
    const val LOVE = "Love"
    const val BAN = "Ban"
    const val NORMAL = "Normal"

    @kotlin.annotation.Retention(AnnotationRetention.SOURCE)
    @StringDef(BAN, LOVE, NORMAL)
    annotation class State
}
